script.js

```diff
@@ -1 +1,402 @@
+let timerId; // to store ID of startMatchTimer
+
+let gameStatus = 'playGame';
+
+// to store the square elements that the player clicked on this round
+let firstClickedSquare;
+let secondClickedSquare;
+
+// player cant click on squares at start of game
+// he or she must submit their name first
+let canClick = false;
+
+const gameInfo = document.createElement('div');
+
+// container to display score message
+const scoreMessage = document.createElement('div');
+
+const playerNameInput = document.createElement('input');
+const playerNameButton = document.createElement('button');
+// Container for player name input and button
+const playerNameInputContainer = document.createElement('div');
+
+// container to display match's countdown timer
+const matchCountdownTimer = document.createElement('div');
+
+// Helper functions -----------------------------------------------
+// to start match timer
+const startMatchTimer = (gameInitNextGame, makeDeck, shuffleCards, buildBoardElements) => {
+  const matchTimerId = setInterval(() => {
+    secondsLeftTillGameOver -= 1;
+    matchCountdownTimer.innerText = `Time Left: ${secondsLeftTillGameOver}s`;
+
+    if (secondsLeftTillGameOver <= 0) {
+      clearInterval(matchTimerId);
+      gameStatus = 'stopGame';
+      numOfGames += 1;
+      gameInitNextGame(makeDeck, shuffleCards, buildBoardElements);
+    }
+  }, 1000);
+
+  return matchTimerId;
+};
+
+// to initilize the next game when game timer runs out
+const gameInitNextGame = (makeDeck, shuffleCards, buildBoardElements) => {
+  // clear the browser content
+  document.body.innerHTML = '';
+
+  // empty the current board
+  board = [];
+
+  // create this special deck by getting the doubled cards and
+  // making a smaller array that is ( boardSize squared ) number of cards
+  const doubleDeck = makeDeck(); // must not shuffle this deck bc you want to take out the pairs
+  const deckSubset = doubleDeck.slice(0, boardSize * boardSize);
+  deck = shuffleCards(deckSubset);
+
+  // deal the cards out to the new board data structure
+  for (let i = 0; i < boardSize; i += 1) {
+    board.push([]);
+    for (let j = 0; j < boardSize; j += 1) {
+      board[i].push(deck.pop());
+    }
+  }
+
+  const boardEl = buildBoardElements(board);
+
+  document.body.appendChild(boardEl);
+
+  // start timer till match ends
+  secondsLeftTillGameOver = 180;
+
+  timerId = startMatchTimer(gameInitNextGame, makeDeck, shuffleCards, buildBoardElements);
+
+  // initialize game info div with lose message
+  gameInfo.innerText = loseMessage;
+
+  document.body.appendChild(gameInfo);
+
+  document.body.appendChild(scoreMessage);
+
+  // display match's countdown timer
+  matchCountdownTimer.innerText = `Time Left: ${secondsLeftTillGameOver}s`;
+  document.body.appendChild(matchCountdownTimer);
+
+  // change global variables accordingly -------------------
+  gameStatus = 'playGame';
+  canClick = true;
+  numOfMatches = 0;
```

Comment:
Newlines between each line of code make it slightly harder to read your code, than if you had more logical groupings of your code with newlines between each group.